python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1705,6 +1705,60 @@ def get_materialization_count_by_partition(
 
         return materialization_count_by_partition
 
+    def get_latest_asset_materialization_planned_event_by_partition(
```

Comment:
This function looks a lot like the `get_materialization_count_by_partition`. I think it's worth refactoring the existing method to also contain functionality from this function.

We don't actually use counts, we just want to figure out if a partition is materialized. I think we'll want to refactor the method to return the mapping from `asset_key, partition_key -> latest storage ID`. And we can accept an `event_type` argument so we can get the latest storage ID per partition for materialization and materialization planned events, which should give us the info we need